### PR TITLE
feat(policy): ListNamespaces search

### PR DESCRIPTION
**Pull request by c-r33d**
### Proposed Changes

1.) Add substring searching to ListNamespaces, `fqn` field.
2.) Escape characters used by `LIKE\ILIKE` from incoming input
3.) Specify `\` as the escape character

### Checklist

- [ ] I have added or updated unit tests
- [ ] I have added or updated integration tests (if appropriate)
- [ ] I have added or updated documentation

## Summary by CodeRabbit

* **New Features**
  * Added search functionality to namespace listing with support for name and FQN matching
  * Search is case-insensitive and supports prefix matching
  * Search integrates with namespace state filtering (ACTIVE/INACTIVE)
  * Special wildcard characters are properly escaped to prevent unexpected matches

* **Tests**
  * Added comprehensive test coverage for search functionality, pagination, and edge cases

### CODE REVIEW

**Comment by c-r33d** [comment]
@coderabbitai review

File: service/integration/namespaces_test.go [hunk 1]
```diff
@@ -582,6 +582,191 @@ func (s *NamespacesSuite) Test_ListNamespaces_Offset_Succeeds() {
...
+	s.Require().NoError(err)
+	beta, err := s.db.PolicyClient.CreateNamespace(s.ctx, &namespaces.CreateNamespaceRequest{
+		Name: fmt.Sprintf("dspx-search-beta-%d.com", suffix),
+	})
+	s.Require().NoError(err)
+	other, err := s.db.PolicyClient.CreateNamespace(s.ctx, &namespaces.CreateNamespaceRequest{
+		Name: fmt.Sprintf("dspx-other-%d.com", suffix),
+	})
+	s.Require().NoError(err)
+	defer s.deleteSortTestNamespaces([]string{alpha.GetId(), beta.GetId(), other.GetId()})
```
Comment (elizabethhealy) [inline service/integration/namespaces_test.go]: nit: since the helper is now being used outside of sort tests consider renaming

**Review by elizabethhealy — DISMISSED** [review]
